docs/modules/language-reference/pages/index.adoc

```diff
@@ -4851,6 +4835,161 @@ The predicate, enclosed in double brackets (`\[[...]]`), is matched against each
 Within the predicate, `this` refers to the member that the predicate is matched against.
 Matching members are amended (`{ ... }`) or overridden (`= <new-value>`).
 
+[[this-keyword]]
+=== `this` keyword
+
+Normally, the `this` keyword references the enclosing object's receiver.
+
+Example:
+[source,pkl]
+----
+bird {
+  eatsInsects = this is InsectavorousBird
+}
+----
+
+When used inside a <<type-constraints,type constraint>>, `this` refers to the value being tested.
+
+Example:
+[source,pkl]
+----
+port: UInt16(this > 1000)
+----
+
+When used inside a <<member-predicates,member predicate>>, `this` refers to the value being matched against.
+
+Example:
+[source,pkl]
+----
+animals {
+  [[this is Bird]] {
+    canFly = true
+  }
+}
+----
+
+[[receiver]]
+==== Receiver
+
+The receiver is the bottom-most object in the <<prototype-chain>>.
+That means that, within the context of an amending object, the reciever is the amending object.
+
+Example:
+[source,pkl]
+----
+hidden lawyerBird {
+  title = "\(this.name), Esq."
+}
+
+polly = (lawyerBird) {
+  name = "Polly" // <1>
+}
+----
+<1> Polly has title `"Polly, Esq."`.
+
+[[outer-keyword]]
+=== `outer` keyword
+
+The `outer` keyword references the <<receiver,receiver>> of the outer lexical object.
```

Comment:
Is this right?
```suggestion
The `outer` keyword references the <<receiver,receiver>> of the immediately outer lexical object.
```

In the past I think I've expected the normal name resolution behavior but _starting at_ the next outer lexical object, but was surprised that it was _only_ the next outer lexical object.

Reply:
Yeah, this is correct. Good clarification!